File: backend/src/main/java/ipeps/pwd/wallet/payload/updatePayload/DocumentUpdatePayload.java
package ipeps.pwd.wallet.payload.updatePayload;

import ipeps.pwd.wallet.entity.Employee;
import ipeps.pwd.wallet.entity.Organization;
import ipeps.pwd.wallet.entity.Transaction;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor

public class DocumentUpdatePayload {
    private UUID document_id;
    private String name;
    private String description;
    private String path;
    private String type;
    private boolean freeAccess;
    private String transaction_id;
    private String employee_id;
    private Organization organization_id;
}

File: frontend/src/app/modules/document/Model/payload/document-update.payload.ts
import { PayloadInterface } from "@shared/model";
import { EmployeeDto } from '@employee/model';
import { OrganizationDto } from '@organization/model';

export interface DocumentUpdatePayload extends PayloadInterface {
  document_id: string;
  name: string;
  description: string;
  path: string;
  type: string;
  freeAccess: boolean;
  employee: EmployeeDto;
  organization: OrganizationDto;
}
